lib/widgets/content.dart

```diff
@@ -804,11 +806,148 @@ class MathBlock extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     final contentTheme = ContentTheme.of(context);
-    return _CodeBlockContainer(
-      borderColor: contentTheme.colorMathBlockBorder,
-      child: Text.rich(TextSpan(
-        style: contentTheme.codeBlockTextStyles.plain,
-        children: [TextSpan(text: node.texSource)])));
+
+    final nodes = node.nodes;
+    if (nodes == null) {
+      return _CodeBlockContainer(
+        borderColor: contentTheme.colorMathBlockBorder,
+        child: Text.rich(TextSpan(
+          style: contentTheme.codeBlockTextStyles.plain,
+          children: [TextSpan(text: node.texSource)])));
+    }
+
+    return _Katex(inline: false, nodes: nodes);
+  }
+}
+
+// Base text style from .katex class in katex.scss :
+//   https://github.com/KaTeX/KaTeX/blob/613c3da8/src/styles/katex.scss#L13-L15
+const kBaseKatexTextStyle = TextStyle(
+  fontSize: kBaseFontSize * 1.21,
```

Comment:
A separate bug: this doesn't correctly handle scaling the font size up when the TeX appears inside a heading. For example: [#test here > math @ 💬](https://chat.zulip.org/#narrow/channel/7-test-here/topic/math/near/2159072)

To do that, this factor of 1.21 should instead get applied in the same way as the scaling factor for the font size of code spans. See references to `kInlineCodeFontSizeFactor`.

Let's make that fix as a separate commit, though, which can go in a follow-up PR. That way we can still merge this PR's changes very soon, as a baseline for further progress.

Reply:
Added a fix for this in ffcdfb0c83c8a8083bc827152a93141d6ed7bce9, in #1452.